vllm/model_executor/models/utils.py

```diff
@@ -222,11 +224,15 @@ def load_weights(
         weights: Iterable[Tuple[str, torch.Tensor]],
         *,
         mapper: Optional[WeightsMapper] = None,
-    ) -> List[str]:
+    ) -> Set[str]:
         if mapper is not None:
             weights = mapper.apply(weights)
 
-        autoloaded_weights = list(self._load_module("", self.module, weights))
+        autoloaded_weights = set(self._load_module("", self.module, weights))
+        if (weights_not_loaded := self.weights_to_load - autoloaded_weights):
+            logger.warning(
+                "Following weights were not initialized from checkpoint: %s",
+                weights_not_loaded)
```

Comment:
Can we have a setting (like `strict=True`) to raise an error when any weights are not loaded? Then we can robustly test the weight loading of all models in `tests/models/test_initialization.py`.